xbmc/cores/AudioEngine/Sinks/pipewire/AESinkPipewire.cpp

```diff
@@ -477,6 +477,22 @@ unsigned int CAESinkPipewire::AddPackets(uint8_t** data, unsigned int frames, un
 
   stream->QueueBuffer(pwBuffer);
 
+  do
+  {
+    pw_time time = stream->GetTime();
+
+    const std::chrono::duration<double, std::ratio<1>> delay =
+        PWTimeToAEDelay(time, m_format.m_sampleRate);
+
+    if (delay < m_period)
```

Comment:
if (delay < (m_buffer_time - m_period))

You want to wait until you can fit a complete period into your buffer without increasing the buffer further. Your condition will cause to not have any time more than 50 ms in the buffer at all, which makes it very underrun prone.

Reply:
updated